pages/admin/treatments/components/TreatmentModal.tsx

```diff
@@ -0,0 +1,252 @@
+import ErrorMessage from '@/components/ErrorMessage';
+import { Treatment } from '@/interfaces/treatment/Treatment';
+import { validateTreatmentFields } from '@/utils/treatments/treatmentValidation';
+import {
+    Box,
+    Button,
+    CloseButton,
+    Dialog,
+    DialogPositioner,
+    Field,
+    Input,
+    Portal,
+    Text,
+} from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+
+interface TreatmentModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    mode: 'create' | 'edit' | 'cancel' | 'delete';
+    selectedTreatment?: Treatment;
+    onTreatmentCreated?: () => void;
+    onSubmit: (treatmentData?: Omit<Treatment, 'id'>) => Promise<void>;
+    isSubmitting: boolean;
+}
+
+export default function TreatmentModal({
+    isOpen,
+    onClose,
+    mode,
+    selectedTreatment,
+    onSubmit,
+    isSubmitting,
+}: TreatmentModalProps) {
+    const [formData, setFormData] = useState({
+        name: '',
+        description: '',
+        price: 0,
+        duration: 0,
+    });
+    const [errors, setErrors] = useState<Record<string, string>>({});
+
+    useEffect(() => {
+        if (isOpen) {
+            setErrors({});
+        }
+        if (isOpen && mode === 'edit' && selectedTreatment) {
+            setFormData({
+                name: selectedTreatment.name,
+                description: selectedTreatment.description,
+                price: selectedTreatment.price,
+                duration: selectedTreatment.duration,
+            });
+            return;
+        }
+        if (mode === 'create') {
+            setFormData({
+                name: '',
+                description: '',
+                price: 0,
+                duration: 0,
+            });
+        }
+    }, [isOpen, mode, selectedTreatment]);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value, type } = e.target;
+        // clean errors
+        setErrors((prevErrors) => {
+            const newErrors = { ...prevErrors };
+            delete newErrors[name];
+            return newErrors;
+        });
+        setFormData({
+            ...formData,
+            [name]: type === 'number' ? Number(value) : value,
+        });
+    };
+
+    const handleSubmit = () => {
+        if (mode === 'create' || mode === 'edit') {
+            // Field validation
+            const fieldErrors = validateTreatmentFields(formData);
+            if (Object.keys(fieldErrors).length > 0) {
+                setErrors(fieldErrors);
+                return;
+            }
+            onSubmit(formData);
+        } else if (mode === 'delete') {
+            onSubmit();
+        }
+    };
+
+    const getTitle = () => {
```

Comment:
💭 I think you can create a constant with the possible titles like:

```js
const TITLES = {
  create: 'Nuevo tratamiento',
  edit: 'Editar tratamiento'
...
}
```

And then call it this way to avoid the switch:

```js
  TITLES[mode]
```